### PR TITLE
Update AndroidBLEHRProvider.java

### PR DESCRIPTION
**Pull request by sdebaerd**
Removed the Huawei Pattern and logic

### CODE REVIEW

**Comment by gerhardol** [comment]
Will have to resurrect a BLE HR strap and test this...

**Comment by sv2ramanan** [comment]
Maybe in the interest of clarity you can comment out this line (#102) as well:

    static final Pattern huamiPattern = Pattern.compile("(Amazfit +Bip|Mi +.*Band)");

thanks.

**Comment by sdebaerd** [comment]
Updated the patch, thanks for the review.

**Comment by sdebaerd** [comment]
If it all works fine, I propose to delete the commented lines?

**Comment by sv2ramanan** [comment]
I checked the PR against (a) Amazfit Bip S (paired) and (b) Decathlon Kalenji dual chest strap (BLE) and both worked fine in the app.

**Review by gerhardol — COMMENTED** [review]
Add "Fixes #1066" to the description.

With this, all BLE devices that are paired are listed, also if they are not connected, which is a little confusing. (Regardless if they have HR service?) These should probably be marked in the list somehow. Maybe add " (paired)" to the name at least.

In addition, I can see my Suunto Ambit3 that do not send HR, only receives (it is paired for notifications). Extra confusing since it has HR. It obviously shows nothing if connecting, the device is not stored when leaving settings.
Also something that calls for a special display of these devices.

I suggest some printouts as a start - limits confusion from hours to minutes.

File: hrdevice/src/org/runnerup/hr/AndroidBLEHRProvider.java [hunk 1]
```diff
@@ -235,9 +235,9 @@ public void onCharacteristicRead(BluetoothGatt gatt,
                     log("firmware => startHR()");
                     // triggered from DummyReadForSecLevelCheck
                     startHR();
-                } else if (huamiMatcherFound && charUuid.equals(HARDWARE_REVISON_UUID)) {
+                } else if (charUuid.equals(HARDWARE_REVISON_UUID)) {
```
[the remark this answers is not in the document]
Comment (sdebaerd) [reply hrdevice/src/org/runnerup/hr/AndroidBLEHRProvider.java:231]: Changed as per your suggestion

File: hrdevice/src/org/runnerup/hr/AndroidBLEHRProvider.java [hunk 2]
```diff
@@ -538,11 +538,11 @@ public void startScan() {
                     continue;
                 }
                 // Bonded device detected :Amazfit Bip S
-                huamiMatcherFound = huamiPattern.matcher(btDeviceThis.getName()).find();
-                if (huamiMatcherFound) {
-                    log("Bonded Huami device detected :" + btDeviceThis.getName());
+                //huamiMatcherFound = huamiPattern.matcher(btDeviceThis.getName()).find();
+                //if (huamiMatcherFound) {
+                //    log("Bonded Huami device detected :" + btDeviceThis.getName());
                     mLeScanCallback.onLeScan(btDeviceThis, 0, null);
```
[the remark this answers is not in the document]
Comment (sdebaerd) [reply hrdevice/src/org/runnerup/hr/AndroidBLEHRProvider.java]: Good call, added the logging

File: hrdevice/src/org/runnerup/hr/AndroidBLEHRProvider.java [hunk 3]
```diff
@@ -538,11 +538,11 @@ public void startScan() {
                     continue;
                 }
                 // Bonded device detected :Amazfit Bip S
```
[the remark this answers is not in the document]
Comment (gerhardol) [reply hrdevice/src/org/runnerup/hr/AndroidBLEHRProvider.java]: adding this in a followup

**Comment by gerhardol** [comment]
There seem to be no quick fix for renaming the devices, it seems to be easier to separate paired and scanned devices be extending AndroidBLEHRProvider.
Just adding a comment for now.

**Comment by gerhardol** [comment]
Thanks!

**Comment by gerhardol** [comment]
This PR breaks Pebble functionality.

Options forward:
1. Revert this and #1027 to get forward, unless this can be resolved quickly.
2. Add option like "experimental devices" to enable this functionality. Will still break Pebble (popup?) and limits the disverability.
3. Better detection, not blocking Pebble and not confusing users for unrelated devices.

**Comment by gerhardol** [comment]
Implemented according to 2.
The functionality need to be enabled from the menu, this was the "safe" implementation.

Cannot test that Huami etc (or Pebble for that matter) still works.
Please try master, release on GitHub or Play beta to verify that Huami etc still works.

**Comment by sv2ramanan** [comment]
I can confirm that the huami (Amzfit) Bip S works with the 2.4.4.0 release, if and only if the "Paired" in the menu is enabled. Unfortunately, I dont have a Pebble.

**Pull request merged**